src/array_string.rs

```diff
@@ -322,14 +322,16 @@ impl<A: Array<Item=u8>> ArrayString<A> {
 
     /// Set the strings's length.
     ///
-    /// May panic if `length` is greater than the capacity.
-    ///
     /// This function is `unsafe` because it changes the notion of the
     /// number of “valid” bytes in the string. Use with care.
+    ///
+    /// This method uses a *debug assertion* to check the validity of
+    /// `length` (character boundary and inside capacity).
     #[inline]
     pub unsafe fn set_len(&mut self, length: usize) {
         debug_assert!(length <= self.capacity());
         self.len = Index::from(length);
+        debug_assert!(self.is_char_boundary(length));
```

Comment:
Oh sigh this is actually broken. `self as &str` depends on length, and the current length is always a character boundary...